src/outputs/frozen_group.py

```diff
@@ -0,0 +1,129 @@
+from src.staging.validation import load_schema
+
+OutputMainLogger = logging.getLogger(__name__)
+
+
+def output_frozen_group(
+    df_gb: pd.DataFrame,
+    df_ni: pd.DataFrame,
+    ultfoc_mapper: pd.DataFrame,
+    config: Dict[str, Any],
+    write_csv: Callable,
+    run_id: int,
+    deduplicate: bool = True
+):
+    """Creates a "frozen group" output  for the entire UK. In BERD (GB) data, 
+    creates foreign ownership and cora status. Selects the columns we need for
+    this type of output. Combines BERD and NI data. Adds size bands. De-
+    duplicates the dataframe, combining the values of multiple sites in one
+    record. Adds blank columns and zero columns - they need to be in the output
+    for back-compatibility, but we don't have any data. output CSV files.
+
+    Args:
+        df_gb (pd.DataFrame): The GB microdata with weights applied
+        df_ ni (pd.DataFrame): The NI microdata; weights are 1
+        ultfoc_mappe (pd.DataFrame): Ultimate foreign owner mappper.
+        config (dict): The configuration settings.
+        write_csv (Callable): Function to write to a csv file.
+         This will be the hdfs or network version depending on settings.
+        run_id (int): The current run id
+        deduplicate (bool): If true, the data is deduplicated by aggregation.
+
+    """
+
+    NETWORK_OR_HDFS = config["global"]["network_or_hdfs"]
+    paths = config[f"{NETWORK_OR_HDFS}_paths"]
+    output_path = paths["output_path"]
+
+    # Categorical columns that we have in BERD and NI data
+    category_columns = [
+        "period_year", "reference","200", "201", "formtype",
+        "employment", "ultfoc", "form_status",
+        "wowenterprisereference", "rusic",
+    ]
+
+    # Numerical value columns that we have in BERD and NI data
+    value_columns = [
+        "emp_researcher", "emp_technician", "emp_other",
+        "emp_total", "202", "203", "204", "205", "206", "207", "209", "210",
+        "211", "212", "214", "216", "218", "219", "220", "221", "222",
+        "223", "225", "226", "227", "228", "229", "237", "242", "243", "244",
+        "245", "246", "247", "248", "249", "302", "303", "304", "305",
+        "headcount_res_m", "headcount_res_f", "headcount_tec_m",
+        "headcount_tec_f", "headcount_oth_m", "headcount_oth_f",
+        "headcount_total", "250", "251", "307", "308", "309",
+    ]
+
+    # Columns that we don't have that should have pd.NA values
+    blank_columns = [
+        "freeze_id", "period", "cell_id", "period_contributor_id",
+    ]
+
+    # Columns that we don't have that should have zero values
+    # The numerical questions starting with q, like q208, are needed for the
+    # output. If it's without q, then we have it in our data.
+    zero_columns = [
+        "data_source", "q208", "q213", "q215", "q217", "q224", "q230", "q231", "q232",
+        "q233", "q234", "q235", "q236", "q238", "q239", "q240", "q241", "q252",
+        "q253", "q254", "q255", "q256", "q257", "q258",
```

Comment:
There are a couple of discrepancies between the file with the output format (link in JIRA) and the AC detailed in the ticket - just want to check these.
In the ticket, under AC4 it says: Blanks (very few with return data): q251, q307, q308, q309, q252.
Looking at the output the values of q251, q307, q308, and q309 are being taken from there respective cols, but don't contain any blanks (they contain a mix of numbers and yes/no) 
q252 has been filled with zero - which is correct according to file.

Just wanted to check these